agent/src/main/scala/mesos/Import.scala

```diff
@@ -102,18 +103,23 @@ object Import {
   /**
    * check if field exists and it's set to 1
    */
-  private[mesos] def checkpasses(json: Json, checkfield: String): Boolean = {
-
-    implicit def metricsDecodeJson: DecodeJson[MetricsChecker] =
-      DecodeJson(c => for {
-        field <- (c --\ checkfield).as[Int]
-      } yield MetricsChecker(field))
-
-    val jo = json.jdecode(metricsDecodeJson)
-
-    jo.value match {
-      case Some(MetricsChecker(1)) => true
-      case _ => false
+  private[mesos] def checkpasses(json: Json, checkfield: Option[String]): Boolean = {
+
+    checkfield match {
+      case None => true
+      case _ => {
+        implicit def metricsDecodeJson: DecodeJson[MetricsChecker] =
+          DecodeJson(c => for {
+            field <- (c --\ checkfield.get).as[Int]
```

Comment:
please lets never call Option.get, instead make it

    case Some(cf) => ...

and then you can use `cf` instead of `checkfield.get`